eng/pipelines/templates/steps/vcpkg.yml

```
@@ -33,6 +33,7 @@ steps:
         $(Agent.JobName)
         | "$(VcpkgCommit)"
         | $(Agent.Os)
+        | variables['${{ parameters.DependenciesVariableName }}']
```

This is the right way to go. Our cache will be more useful this way if the dependencies vary based on the job executing. 

However, I think this only outputs the literal string: 

`| variables['vcpkg.deps']` 

To get this to output the list of dependencies we should expand the variable using [macro syntax](https://docs.microsoft.com/en-us/azure/devops/pipelines/process/variables?view=azure-devops&tabs=yaml%2Cbatch#macro-syntax-variables) as it is an input to a task. 

```suggestion 
        | $(${{ parameters.DependenciesVariableName }})
```

This syntax is annoying to read but it basically says: 

`${{ parameters.DependenciesVariableName }}` -- Before expanding the YAML into steps, rewrite the YAML with this string replaced with the parameter value. Result: `vcpkg.deps` (or whatever is set for that parameter) 

`$(x)` -- At runtime, substitute the variable whose name is `x` into the string that is being built for this task. 

Since the `${{ parameters.___ }}` syntax mutates the YAML that will be processed we'll end up writing the variable name into the macro syntax which will then insert the variable value in the string.

---

That makes sense to me, let me try it and see what it does.